pkg/server/service/etcdproxy/etcd_proxy.go

```diff
@@ -209,12 +209,16 @@ func (e *etcdProxy) Txn(ctx context.Context, txn *etcdserverpb.TxnRequest) (*etc
 	}
 	if !resp.GetSucceeded() {
 		getResp := (*clientv3.GetResponse)(resp.Responses[0].GetResponseRange())
```

Comment:
should check len of `resp.Responses` too？

Reply:
Yes, it is. The leader's code responds to normal requests with responses, but in terms of defensive programming, it really needs to be judged